src/connection-state-handler.ts

```diff
@@ -113,6 +122,32 @@ export class ConnectionStateHandler extends EventEmitter<ConnectionStateEventHan
    * @returns Current connection state.
    */
   public getConnectionState(): ConnectionState {
-    return this.mediaConnectionState;
+    const { connectionState } = this.getCurrentStatesCallback();
+
+    const state = connectionState[0].toUpperCase() + connectionState.slice(1);
+
+    return state as ConnectionState;
+  }
+
+  /**
+   * Gets current ice connection state.
+   *
+   * @returns Current ice connection state.
+   */
+  public getIceConnectionState(): IceConnectionState {
+    const { iceState } = this.getCurrentStatesCallback();
+
+    const state = iceState[0].toUpperCase() + iceState.slice(1);
+
+    return state as IceConnectionState;
+  }
+
+  /**
+   * Gets current overall connection state.
+   *
+   * @returns Current overall connection state.
+   */
+  public getOverallConnectionState(): OverallConnectionState {
```

Comment:
Wondering if it makes sense to keep calling this the `connectionState` and have the 2 sub-states be called `peerConnectionState` and `iceConnectionState`?

Reply:
Changed it as suggested. I like it, however had to change event name to keep it consistent. What do you think about it?